[김강우] 생성된 롤링페이퍼 페이지 "post/{id}"

hvrain opened the pull request:
기본 요구 사항

- [x] 카드 목록은 '최신순'으로 무한 스크롤 방식으로 배치합니다.
- [x] '카드'를 누르면 해당 카드가 확대되어 보여집니다.
- [x] PC에서 헤더 영역의 너비는 1200px로 고정하고 좌우 여백만 변하다가, 화면의 너비가 1248px보다 줄어들 때 좌우 여백은 24px로 고정하고 'To.OOO' 영역과 '프로필 이미지 OO명이 작성 했어요!' 영역이 가까워집니다.
- [ ] PC에서 카드 목록 영역의 너비는 1200px로 고정하고 좌우 여백만 변하다가, 화면의 너비가 1248px보다 줄어들 때 좌우 여백은 24px로 고정하고 카드들의 너비가 줄어듭니다.
- [x] 저장된 이모지는 횟수가 많은 순서대로 최대 '6개' 까지 나열해 주세요.
- [x] [[카카오톡 공유 docs](https://developers.kakao.com/docs/latest/ko/message/js-link)](https://developers.kakao.com/docs/latest/ko/message/js-link)를 참고해 '카카오톡 공유' 기능을 만들어 주세요.
(구현 방법이나 공유에 들어갈 메시지는 팀 내에서 논의해 결정해 주세요)
- [x] 'URL 공유'를 클릭하면 클립보드에 URL을 저장하고, URL이 복사되었다는 
토스트를 5초 동안 보였다가 사라지게 해주세요.
- [x] 이모지 추가 드롭다운 디자인은 예시이고, 이와 동일한 기능을 하는 외부 모듈의 디자인을 
그대로 사용해도 좋습니다.
- [x] '+' 버튼을 누르면 `/post/{id}/message` 로 이동합니다.

추가 구현 사항

- [x] 카드에 마우스를 hover하면, 그림자를 진하게 만들었습니다.

리뷰어에게

- 기본 요구 사항에 카드 목록 영역 반응형 구현을 다르게 했습니다. 요구 사항을 늦게 봐서 잘못 구현했는데, 지금 고치려면 시간이 많이 들어서 나중에 고치면 commit해서 올리겠습니다.

hvrain committed:
Add: postRecipientMessage function in dummyPost.js

hvrain committed:
Finish setup dummy api

hvrain committed:
Add: fetch getRecipientById & Use in PostId.jsx

hvrain committed:
Install: styled-components

hvrain committed:
Add: getSendersProfile in PostId.jsx

hvrain committed:
Add: postRecipientsReactions in dummyPost.js

hvrain committed:
Update: postRecipientMessage in dummyPost.js

hvrain committed:
Add: api get from Promise.all

hvrain committed:
Add: throttle function in utils

hvrain committed:
Add: 무한 스크롤 (scroll 이벤트, throttle)버전

hvrain committed:
Update: 무한 스크롤 (update scroll event to IntersectionObserver)

hvrain committed:
Eslint rule off (react/prop-types)

hvrain committed:
Add: styled-components ThemeProvider and theme.js

hvrain committed:
Add: SenderProfile.jsx

hvrain committed:
Add: axios config

hvrain committed:
Add: Reactions.jsx and Shared.jsx

hvrain committed:
Add: Modal.jsx

hvrain committed:
push용 임시커밋

hvrain committed:
Add: Toast.jsx and use in Shared.jsx

hvrain committed:
Add: dropdown manage context in RecipientHeadereProvider.jsx"

이전엔 3개의 다른 드롭다운 메뉴들을 각각의 state값으로 제어했었다.
수정을 하여 하나의 state값으로 제어하며, 단 하나의 함수로 전체 dropdown을 관리하는 Provider를 만들어 사용하고 있다.

hvrain committed:
Add: font 동적 할당 적용

서버에서 주어지는 4가지 글씨체에 해당하는 폰트를 cdn으로 불러와 styled-component로 적용

hvrain committed:
Add: postReactionsByRecipientId

hvrain committed:
Add: PostHeader.jsx

hvrain committed:
Edit: Emoji picker style and Remove: console.log

hvrain committed:
Edit: Emoji picker style and Remove: console.log

hvrain committed:
Merge branch 'feature/postId' of https://github.com/codeit-basic-project-8-5/rolling into feature/postId

hvrain committed:
Add: link post/{id}/message

ayoooyh commented:
수정하다 강우님 브랜치 pull해서 수정중인데 이전에 styles/styled/에 있던 Reactions 파일이 없네요..! 그 파일에 있던 BorderButton을 사용해서 button 만들었었는데 파일명이 바뀐건지 아예 삭제하신건지 알려주세요!!

hvrain commented:
> 수정하다 강우님 브랜치 pull해서 수정중인데 이전에 styles/styled/에 있던 Reactions 파일이 없네요..! 그 파일에 있던 BorderButton을 사용해서 button 만들었었는데 파일명이 바뀐건지 아예 삭제하신건지 알려주세요!!

styles/styled/PostId.js 에 있을 겁니다

hvrain committed:
Fix: RecipientHeader 페이지 수정

RecipientHeader 컴포넌트의 내용을 수정하고, 그에 따라 안에 들어가는 스타일 컴포넌트의 css수정

hvrain committed:
Fix: styled-components import 방식 변경

import 할때 각각의 컴포넌트를 하나하나 불러왔었음
변경 후 import * as S 방식으로 변경하여, react 컴포넌트랑 구별되게 만듦

hvrain committed:
Fix: 7084fc8 커밋의 에러 수정

해당 커밋 이후 mobile에서 RecipientHeader의 스타일 문제가 발생해서 컴포넌트 내용 mobile과 mobile이 아닌 것으로 분리함

hvrain committed:
Fix: Toast.jsx의 styled-components import 방식 변경

hvrain committed:
Fix: Modal의 스타일 변경

이전에 Modal이 모바일 환경에서 스타일이 깨지는 상황 발생
수정 후 Modal이 viewport에 따라 width가 바뀌게 함

hvrain committed:
Fix: 무한 스크롤 에러 수정

두번의 Message api 호출동안 페이지의 viewport의 높이만큼 요소를 채우지 못하면 더 이상 호출하지 않는 에러 발생
useEffect에 messages를 의존성에 추가함
수정 후, viewport에 상관없이 화면을 채울만큼 api 호출함
